internal/fs/parallel_dirops_test.go

```diff
@@ -0,0 +1,687 @@
+	assert.NoError(t.T(), err1)
+	assert.NoError(t.T(), err2)
+	assert.Equal(t.T(), int64(6), stat1.Size())
+	assert.Equal(t.T(), int64(3), stat2.Size())
+	assert.Contains(t.T(), filePath1, stat1.Name())
+	assert.Contains(t.T(), filePath2, stat2.Name())
+
+	// Parallel lookups of two files under a directory in mount.
+	filePath1 = path.Join(mntDir, "explicitDir1", "file1.txt")
+	filePath2 = path.Join(mntDir, "explicitDir1", "file2.txt")
+	wg = sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		stat1, err1 = lookUpFunc(&wg, filePath1)
+	}()
+	go func() {
+		stat2, err2 = lookUpFunc(&wg, filePath2)
+	}()
+	wg.Wait()
+
+	// Assert both stats passed and give correct information
+	assert.NoError(t.T(), err1)
+	assert.NoError(t.T(), err2)
+	assert.Equal(t.T(), int64(5), stat1.Size())
+	assert.Equal(t.T(), int64(10), stat2.Size())
+	assert.Contains(t.T(), filePath1, stat1.Name())
+	assert.Contains(t.T(), filePath2, stat2.Name())
+}
+
+func (t *ParallelDiropsTest) TestParallelLookUpsForDifferentDirs() {
+	lookUpFunc := func(wg *sync.WaitGroup, dirPath string) (os.FileInfo, error) {
+		defer wg.Done()
+		fileInfo, err := os.Stat(dirPath)
+		return fileInfo, err
+	}
+	var stat1, stat2 os.FileInfo
+	var err1, err2 error
+
+	// Parallel lookups of two dirs under mount.
+	dirPath1 := path.Join(mntDir, "explicitDir1")
+	dirPath2 := path.Join(mntDir, "implicitDir1")
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		stat1, err1 = lookUpFunc(&wg, dirPath1)
+	}()
+	go func() {
+		stat2, err2 = lookUpFunc(&wg, dirPath2)
+	}()
+	wg.Wait()
+
+	// Assert both stats passed and give correct information
+	assert.NoError(t.T(), err1)
+	assert.NoError(t.T(), err2)
+	assert.True(t.T(), stat1.IsDir())
+	assert.True(t.T(), stat2.IsDir())
+	assert.Contains(t.T(), dirPath1, stat1.Name())
+	assert.Contains(t.T(), dirPath2, stat2.Name())
+}
+
+func (t *ParallelDiropsTest) TestParallelReadDirsForDifferentDirs() {
+	readDirFunc := func(wg *sync.WaitGroup, dirPath string) ([]os.DirEntry, error) {
+		defer wg.Done()
+		dirEntries, err := os.ReadDir(dirPath)
+		return dirEntries, err
+	}
+	var dirEntries1, dirEntries2 []os.DirEntry
+	var err1, err2 error
+
+	// Parallel readDirs of explicit dir under mount.
+	dirPath1 := path.Join(mntDir, "explicitDir1")
+	dirPath2 := path.Join(mntDir, "implicitDir1")
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		dirEntries1, err1 = readDirFunc(&wg, dirPath1)
+	}()
+	go func() {
+		dirEntries2, err2 = readDirFunc(&wg, dirPath2)
+	}()
+	wg.Wait()
+
+	// Assert both readDirs passed and give correct information
+	assert.NoError(t.T(), err1)
+	assert.NoError(t.T(), err2)
+	assert.Contains(t.T(), dirEntries1[0].Name(), "file1.txt")
+	assert.Contains(t.T(), dirEntries1[1].Name(), "file2.txt")
+	assert.Contains(t.T(), dirEntries2[0].Name(), "file1.txt")
+}
+
+func TestParallelDiropsTestSuite(t *testing.T) {
+	suite.Run(t, new(ParallelDiropsTest))
+}
+
+func TestParallelDiropsWithoutCachesTestSuite(t *testing.T) {
```

Comment:
Can we keep all the constructs of `WithoutCachesTest` together? I mean, Test struct, SetupTestSuites etc.